src/CommunityToolkit.Maui.Core/Essentials/FileSaver/FileSaverImplementation.android.cs

```diff
@@ -82,7 +85,18 @@ static async Task<string> SaveDocument(AndroidUri uri, Stream stream, Cancellati
 		await using var memoryStream = new MemoryStream();
 		await stream.CopyToAsync(memoryStream, cancellationToken).ConfigureAwait(false);
 		await fileOutputStream.WriteAsync(memoryStream.ToArray()).WaitAsync(cancellationToken).ConfigureAwait(false);
-		var split = uri.Path?.Split(':') ?? throw new FolderPickerException("Unable to resolve path.");
-		return $"{Android.OS.Environment.ExternalStorageDirectory}/{split[^1]}";
+		var path = uri.PathSegments?[1] ?? throw new FolderPickerException("Unable to resolve path.");
+		var split = path.Split(':');
+		// Primary is the device's internal storage, and anything else is an SD card or other external storage
+		if (split[0].Equals("primary", StringComparison.OrdinalIgnoreCase))
```

Comment:
add check that the element exists

Reply:
does constant exist for this value?

Reply:
No constant available, the internal storage is called primary on uri per convention.

Reply:
Let's create a constant PrimaryStorage

Reply:
Done